### ISSUE TITLE
Markdown support

### ISSUE THREAD

**Issue by MHOOO** [gristlabs/grist-core#168]
I wonder, would you be interested in markdown support?
I was thinking of applying a [markdown renderer](https://github.com/markedjs/marked) to text fields if the user chooses a new „markdown“ cell format. What do you think?

If you like the idea, how would you approach the implementation?

**Comment by paulfitz** [comment]
Yes, we're interested in markdown and rich text (see https://github.com/gristlabs/grist-core/issues/67). There is a primitive version of markdown support available via a custom widget:
![Screenshot from 2022-03-15 10-09-32](https://user-images.githubusercontent.com/118367/158396249-54c1071d-0a4f-43dc-8b86-261352565a56.png)
https://docs.getgrist.com/u5rgvdX4V2G9/Markdown-Example/m/fork

Native support would be a lot cleaner.

One question that I'd ask early is how would embedded media (images, animations) be handled, is that in scope for what you are thinking about or not? It can be an awkward corner where markdown gets ugly. Grist has an attachment column where each cell can have any number of attachments, so perhaps that kind of column could be a good "back end" for markdown that has an easy way to refer to images without messy absolute urls?

**Comment by dumblob** [comment]
I see a huge risk here in future additions to this functionality. I guarantee that customers **will** come and will make such an enormous pressure that you'll have to implement it in one way or another. It starts with tables, then with interactive tables, additional styling, interactive media etc. Been there, done that.

All in all this rapidly converges to general complex input as discussed in https://github.com/gristlabs/grist-core/issues/161 .

So my approach would be two fold (both at the same time):

1. Either leave cells as plain text as GitHub does (and maybe add a preview pane or widget following cell focus) or leave everything as plain text but with some very minor "automated" formatting **without** hiding any formatting characters - i.e. `**bold**` will be rendered as **\*\*bold\*\*** (note the asterisk stayed as if it was only plain text) - same for italics, and titles (probably nothing else as it would break the plain text - e.g. any indentation, font size etc. is undesirable).

2. Take the [seamless widget support](https://github.com/gristlabs/grist-core/issues/161 ) path (that overlaps with https://github.com/gristlabs/grist-core/issues/37#issuecomment-1048625090 ) and there you can go wild with much more than just CommonMark.
    
    Otherwise the gap between widgets and cell types would increase rapidly over time without any seamless way to bridge them. Therefore my proposal of validators (but I don't claim this is the only way to avoid the gap).

**Comment by tayflo** [comment]
For reference, I guess this issue could be closed, as in-cell Markdown is now supported (292c894b931e4bba77b1ba6ef6a2a625f4704564).

I however note that, as mentionned in the [docs](https://support.getgrist.com/col-types/#markdown), the current implementation does not currently handle media (those were mentioned [right above](https://github.com/gristlabs/grist-core/issues/168#issuecomment-1068056953) by @paulfitz), neither custom HTLM.

On the same topic, see also #1161 and the [proposal document](https://docs.google.com/document/d/1-NmB880GjtsyILz2QYkHEHclvC8038xt537XpAqA5NY/edit?tab=t.0) in it, which exposes the reasoning behind limited support of the Markdown specification for now.
